cmd/subnetcmd/upgradecmd/generate.go

```diff
@@ -315,6 +324,80 @@ func promptTxAllowListParams(precompiles *[]params.PrecompileUpgrade, date time.
 	return nil
 }
 
+func getCClient(apiEndpoint string, blockchainID string) (ethclient.Client, error) {
+	cClient, err := ethclient.Dial(fmt.Sprintf("%s/ext/bc/%s/rpc", apiEndpoint, blockchainID))
+	if err != nil {
+		return nil, err
+	}
+	return cClient, nil
+}
+
+func ensureAdminsHaveBalanceLocalNetwork(admins []common.Address, blockchainID string) error {
+	cClient, err := getCClient(constants.LocalAPIEndpoint, blockchainID)
+	if err != nil {
+		return err
+	}
+
+	for _, admin := range admins {
+		// we can break at the first admin who has a non-zero balance
+		accountBalance, err := getAccountBalance(context.Background(), cClient, admin.String())
+		if err != nil {
+			return err
+		}
+		if accountBalance > float64(0) {
```

Comment:
This is well thought but my suggestion is that the balance should be at least the minimum transaction fee. Above zero but below the fee will still result in a failure.

Frankly even just one tx fee is probably too small (the user most probably will want to ensure that more than just one tx goes through), but above the minimum transaction fee, we would have to arbitrarily set a minimum. We can quickly touch on this in standup and / or slack if you like.

Reply:
I think this is a good idea, but it might be a little tricky to implement. We'd need to parse gas configs, doable but I'd be willing to backlog this